src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,45 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+
+        if (object == null) {
+            return false;
+        }
+
+        if (object.getClass() == Rectangle.class) {
+            Rectangle rectangle = (Rectangle) object;
+            if (!this.hasAllEmptyParameters()
+                    && !rectangle.hasAllEmptyParameters()) {
+                return width.equals(rectangle.width)
+                        && length.equals(rectangle.length)
+                        && color.equals(rectangle.color);
+            }
+            return this.hasAllEmptyParameters()
+                    && rectangle.hasAllEmptyParameters();
```

Comment:
see checklist: `While comparing fields at equals, use single return statement for all fields(e.g. return check1 && check2 && check3;).`

Reply:
also, I think we don't need to write a helper method